x-pack/plugins/ml/public/explorer/explorer_charts/explorer_charts_container.js

```diff
@@ -7,18 +7,31 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import { EuiIconTip } from '@elastic/eui';
+import {
+  EuiButtonEmpty,
+  EuiIconTip,
+  EuiToolTip
+} from '@elastic/eui';
 
-import { getExploreSeriesLink } from '../../util/chart_utils';
+import {
+  getExploreSeriesLink,
+  isLabelLengthAboveThreshold
+} from '../../util/chart_utils';
 import { ExplorerChart } from './explorer_chart';
-import { ExplorerChartTooltip } from './explorer_chart_tooltip';
+import { ExplorerChartLabel } from './components/explorer_chart_label';
+
+const textTooManyBuckets = 'This selection contains too many buckets to be displayed.' +
+  'The dashboard is best viewed over a shorter time range.';
```

Comment:
Bit of a nitpick... Could probably use template literals here to keep in line with the style guide.